core/src/main/scala/cats/CommutativeApplicative.scala

```diff
@@ -0,0 +1,41 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+  * Commutative Applicative.
+  *
+  * Further than an Applicative, which just allows composition of independent effectful functions,
+  * in a Commutative Applicative those functions can be composed in any order, which guarantees
+  * that their effects do not interfere.
+  *
+  * Must obey the laws defined in cats.laws.CommutativeApplicativeLaws.
+  */
+@typeclass trait CommutativeApplicative[F[_]] extends Applicative[F] with CommutativeApply[F]
+
+
+object CommutativeApplicative {
+
+  def traverseUnordered[F[_]: CommutativeApplicative, A, B](sa: Set[A])(f: A => F[B]): F[Set[B]] =
```

Comment:
Shouldn’t we put these on the trait on the chance they can be optimized for some instances?

Reply:
Good point!